app/knight.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.protection = 0
+
+        self.power += knight["weapon"]["power"]
+
+        for armour in knight["armour"]:
+            self.protection += armour["protection"]
+
+        if knight["potion"]:
+            if "power" in knight["potion"]["effect"]:
+                self.power += knight["potion"]["effect"]["power"]
+
+            if "protection" in knight["potion"]["effect"]:
+                self.protection += knight["potion"]["effect"]["protection"]
+
+            if "hp" in knight["potion"]["effect"]:
+                self.hp += knight["potion"]["effect"]["hp"]
```

Comment:
Use loop with getattr and setattr. No need to duplicate code

Reply:
Sorry)